src/Controls/StreetControls.js

```diff
@@ -77,8 +79,143 @@ class StreetControls extends FirstPersonControls {
         this.previousPosition = newPreviousPosition;
     }
 
+    firstPickedObject(event) {
+        // basic threeJs pick with raycaster
+        const normalized = this.view.viewToNormalizedCoords(event);
+        this.raycaster.setFromCamera(normalized.clone(), this.camera);
+        const intersects = this.raycaster.intersectObjects(this.view.scene.children, true);
+
+        // skipping specific objects:
+        // - skip the two surfaces (rectangle and circle)
+        // - skip all the object without face, because we use the face to manage surface angle, so it's needed.
+        let i = 0;
+        while (i < intersects.length &&
+            (!intersects[i].face || intersects[i].object == this._rectangleMesh || intersects[i].object == this._circleMesh)) {
+            i++;
+        }
+
+        // return first interesting picked object, or undefined
+        if (i < intersects.length) {
+            return intersects[i];
+        }
+    }
+
+    pick(event, pickGround = () => {}, pickObject =  () => {}) {
+        // specific method to pick on the elevation layer at the good elevation, even if elevation layer is done in the shader
+        const mouse = this.view.eventToViewCoords(event);
+        const target = new THREE.Vector3();
```

Comment:
could you move out `target` from this function .

Reply:
ok

Reply:
```js
 pick(event, pickGround = () => {}, pickObject = () => {}, target = new THREE.Vector3())
```
 you don't use `target` apart from the function, you could remove it

Reply:
ok